src/retryLink.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  Operation,
+  NextLink,
+  FetchResult,
+} from './types';
+import * as Observable from 'zen-observable';
+import {
+  ensureForward,
+} from './linkUtils';
+
+import {
+  ApolloLink,
+} from './link';
+
+export default class RetryLink extends ApolloLink {
+
+  private count: number = 0;
+  private delay: number;
+  private max: number;
+  private interval: (delay: number, count: number) => number;
+
+  constructor (params?: {
+    max?: number,
+    delay?: number,
+    interval?: (delay: number, count: number) => number,
+  }) {
+    super();
+    this.max = params && params.max || 10;
+    this.delay =  params && params.delay || 300;
+    this.interval =  params && params.interval || this.defaultInterval;
+  }
+
+  public request(operation: Operation, forward: NextLink): Observable<FetchResult> {
+    ensureForward(forward);
+
+
+    return new Observable(observer => {
+      const subscriber = {
+        next: data => {
+          observer.next(data);
+          this.count = 0;
```

Comment:
I would switch these in case observer.next throws, we wan't the count reset anytime `.next` is called